### PR TITLE
Viewer3D: light position with 2 modes (camera/scence center)

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description

Add a new mode to have the light source position defined from the camera (default) or from the scene coordinate system (key P to change  position mode). A new lambertian rendering mode was added and a fix to the load/save light position.

# Checklist

- [n.a.] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: src/DGtal/io/viewers/Viewer3D.ih [hunk 1]
```diff
@@ -887,6 +886,7 @@ DGtal::Viewer3D<Space, KSpace>::init()
   setKeyDescription ( Qt::Key_D, "Enable/Disable the two side face rendering." );
   setKeyDescription ( Qt::Key_R, "Reset default scale for 3 axes to 1.0f." );
   setKeyDescription ( Qt::Key_M, "Switch the rendering mode bewteen Default, Metallic and Plastic mode." );
+  setKeyDescription ( Qt::Key_P, "Switch the light source position mode between the camera mode (default: the light source position is fix according to the camera position) and scene mode (the light source position is fix according the scene coordinate system)." );
```
Comment (dcoeurjo) [inline src/DGtal/io/viewers/Viewer3D.ih]: is fixed.

**Comment by dcoeurjo** [comment]
The "P" key in my mac does nothing: when I rotate, the light also moves (see for instance testBallQuadViewer). 
Have I missed something?

**Comment by kerautret** [comment]
I have an issue on testBallQuadViewer, for me all the ball is black... Perhaps a pb in the initialization, I look it thanks

**Comment by kerautret** [comment]
ok I got it,... It is a pb of missing default value of camera default position (occurs only when the viewer is initialized outside the main, I don't why, but perhaps related to qt).

**Comment by kerautret** [comment]
@dcoeurjo it should be fine now (I hope) ;)
And now the light source position returns to its default value when we change the mode.

**Comment by dcoeurjo** [comment]
excellent.
it works

**Comment by dcoeurjo** [comment]
Could you please describe this new behavior in the documentation ?

**Comment by kerautret** [comment]
perfect, ok I add more details in the Viewer3D related section of the doxygen doc

**Comment by dcoeurjo** [comment]
great. Just minor thing: would it be possible that the default position of the light is not (0,0,1) (aligned with the view) ?

**Comment by kerautret** [comment]
ok I look it

**Comment by kerautret** [comment]
done for the doc, for the default position, it is not direct due to the potential user light move with the two mode. It could be done later considering camera orientation but perhaps could we merge and add an issue  ? (not a lot of time actually ;( )

**Comment by kerautret** [comment]
oups wait , I found a mistake, in the coordinates change, I have the good solution, i clean and commit it.

File: src/DGtal/io/doc/moduleDisplay3D.dox [hunk 2]
```diff
@@ -92,13 +92,62 @@ You should obtain the following visualization:
...
+
+The light source position is by default defined according to the
+camera position and its position will not change even after camera
+move (default mode). By this way, if you move the camera, the object
+will be always illuminated (see images of mode 0 of the following
+tabular). This default light position could be changed interactively
+by a mouse move in the Viewer3D (with the key SHIFT+CTRL (SHIFT+CMD on
 mac)).
 
+There exists a second mode where the light source position will fixed
```
Comment (dcoeurjo) [inline src/DGtal/io/doc/moduleDisplay3D.dox]: will be fixed

File: src/DGtal/io/doc/moduleDisplay3D.dox [hunk 3]
```diff
@@ -92,13 +92,62 @@ You should obtain the following visualization:
...
+|  3              | lambertian mode     |   ![ ] (lambertianRenderingViewer3Dmode3.png) |
+
+
+
+
+\subsection  DGtalGLV_Viewer3D_Light Light source position modes
+
+The light source position is by default defined according to the
+camera position and its position will not change even after camera
+move (default mode). By this way, if you move the camera, the object
```
Comment (dcoeurjo) [inline src/DGtal/io/doc/moduleDisplay3D.dox]: moves? (plural of the move noun)

**Comment by dcoeurjo** [comment]
Perfect. Than you for the documentation and the edits.
I have minor comments on the doc. You can self-merge if you want when the doc is fixed.

File: src/DGtal/io/doc/moduleDisplay3D.dox (hunk 2 above)
Comment (kerautret) [inline src/DGtal/io/doc/moduleDisplay3D.dox]: thanks

File: src/DGtal/io/doc/moduleDisplay3D.dox (hunk 3 above)
Comment (kerautret) [inline src/DGtal/io/doc/moduleDisplay3D.dox]: yes

**Comment by kerautret** [comment]
thanks ok  I am going to merge, I also remove the note indicating the light will be back to its initial positions by changing mode since now it works well ;)
I ajuste the tabular a little big ;)

**Comment by kerautret** [comment]
merging as suggested by David

**Pull request merged**